### PR TITLE
fix: preserve key order in YAML output

### PR DESCRIPTION
**Pull request by lawrence3699**
Fixes #2028.

**Failing scenario:** Converting JSON to YAML with `mlr --j2y cat` sorts keys alphabetically, while `--j2p` and `--j2j` preserve insertion order.

```sh
$ echo '{"b":22, "c":33, "a":11}' | mlr --j2y cat
- a: 11    # wrong: sorted alphabetically
  b: 22
  c: 33
```

**After this fix:**
```sh
$ echo '{"b":22, "c":33, "a":11}' | mlr --j2y cat
- b: 22    # correct: insertion order preserved
  c: 33
  a: 11
```

**Root cause:** `MlrmapToYAMLNative` converted the ordered `Mlrmap` (doubly-linked list) to `map[string]interface{}`, which has no ordering guarantee. `yaml.v3.Marshal` then sorted map keys alphabetically.

**Fix:** Build `yaml.Node` trees that preserve the `Mlrmap`'s linked-list iteration order. Scalar values use `yaml.Node.Encode` to correctly handle edge cases (NaN/Inf floats, non-UTF-8 strings). Both the list-wrapped and multi-document YAML output paths are updated.

**Validation:**
- All unit tests pass (`go test ./pkg/...`)
- All YAML regression tests pass (`mlr regtest test/cases/io-yaml-io`)
- Added `TestMlrmapToYAMLNativeKeyOrder` to verify insertion-order preservation

### CODE REVIEW

File: pkg/mlrval/mlrval_yaml.go [hunk 1]
```diff
@@ -138,63 +138,79 @@ func mlrvalFromYAMLArray(a []interface{}) (*Mlrval, error) {
...
-// MlrmapToYAMLNative converts an Mlrmap to a Go value suitable for
-// yaml.Marshal: map[string]interface{} with nested maps/arrays.
-func MlrmapToYAMLNative(mlrmap *Mlrmap) (interface{}, error) {
+// MlrmapToYAMLNative converts an Mlrmap to a *yaml.Node that preserves
+// key insertion order. The returned node is suitable for yaml.Marshal.
+func MlrmapToYAMLNative(mlrmap *Mlrmap) (*yaml.Node, error) {
 	if mlrmap == nil {
-		return nil, nil
+		return &yaml.Node{Kind: yaml.MappingNode, Tag: "!!map"}, nil
+	}
```
[the remark this answers is not in the document]
Comment (johnkerl) [reply pkg/mlrval/mlrval_yaml.go:146]: API support was created a few years ago, but always with the understanding that there would be no backward-compatibility guarantees. (Just FYI for the reader.)

**Comment by johnkerl** [comment]
Thanks @lawrence3699 !!! :)
